api/metrics/v1/rules.go

```diff
@@ -71,26 +89,32 @@ func (rh *rulesHandler) get(w http.ResponseWriter, r *http.Request) {
 			switch r := rawRules.Groups[i].Rules[j].(type) {
 			case rules.RecordingRule:
 				if r.Labels.AdditionalProperties == nil {
-					r.Labels.AdditionalProperties = make(map[string]string)
+					http.Error(w, "rule has no label", http.StatusInternalServerError)
```

Comment:
No, you're not missing anything the cases are exactly the same. But we switch by type so that we can get the `Labels`. If we use no switch or use `cases rules.RecordingRule, rules.AlertingRule:` then it's just an interface.

Reply:
Ah, that's an interesting gotcha, now I see, thanks :+1: